File: app/src/main/java/com/example/roomapplication/ResultActivity.java
package com.example.roomapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.example.roomapplication.roomdb.Note;
import com.example.roomapplication.roomdb.NoteDatabase;

import java.util.List;

public class ResultActivity extends AppCompatActivity {

    public List<Note> notes;
    public NoteDatabase noteDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);

        noteDB = NoteDatabase.getInstance(this);
        RecyclerView rvNotes = findViewById(R.id.recyclerView_result);
        notes = noteDB.getNoteDao().getAllNotes();
        NotesAdapter adapter = new NotesAdapter(notes);
        rvNotes.setAdapter(adapter);
        rvNotes.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/java/com/example/roomapplication/ResultActivity.kt
package com.example.roomapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.roomapplication.roomdb.Note
import com.example.roomapplication.roomdb.NoteDatabase
import com.example.roomapplication.roomdb.NoteDatabase.Companion.getInstance

class ResultActivity : AppCompatActivity() {
    var notes: List<Note>? = null
    var noteDB: NoteDatabase? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)
        noteDB = getInstance(this)
        val rvNotes = findViewById<RecyclerView>(R.id.recyclerView_result)
        notes = noteDB!!.noteDao?.allNotes as List<Note>?
        val adapter = NotesAdapter(notes!!)
        rvNotes.adapter = adapter
        rvNotes.layoutManager = LinearLayoutManager(this)
    }
}